test/integration/certificates/metrics_controller_test.go

```diff
@@ -82,7 +79,7 @@ func TestMetricsController(t *testing.T) {
 		}
 	}()
 	defer func() {
-		shutdownCtx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+		shutdownCtx, cancel := context.WithTimeout(context.WithoutCancel(t.Context()), time.Second*5)
```

Comment:
The code in this area appears very complex. Lots of cleanup and defers. Any possibility to simplify? Eventually in a follow-up PR?